src/benchmarks/micro/libraries/System.IO.Compression/Zip.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Linq;
+
+namespace System.IO.Compression
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public class Zip
+    {
+        private static readonly string s_testFoldersLocation = Path.Combine(
+            Path.GetDirectoryName(typeof(Zip).Assembly.Location), "libraries", "System.IO.Compression", "TestData");
+
+        [Params("TestDocument", // 199 KB small test document with repeated paragraph, PDF are common.
+            "alice29", // 145 KB, copy of "ALICE'S ADVENTURES IN WONDERLAND" book, an example of text file.
+            "sum")] // 37.3 KB, some binary content, an example of binary file.
+        public string TestFolder;
+
+        [ParamsAllValues]
+        public CompressionLevel CompressionLevel;
+
+        private ZipArchive _archive;
+        private ZipArchiveEntry _entry;
+        private string _testFileLocation;
+        private string _archiveLocation;
+        // FileName at which the Zip archive entry is extracted to.
+        private string _tempDestinationFileName;
+        // Name of the entry that is created when the file is added into the Zip archive.
+        private string _randomEntryName;
+
+
+        [GlobalSetup(Target = nameof(ExtractEntryToFile))]
+        public void SetupExtractTo()
+        {
+            string currentTestFolderLocation = Path.Combine(s_testFoldersLocation, TestFolder);
+            _archiveLocation = currentTestFolderLocation + "zip";
+
+            // Create a zip archive, then open it.
+            // We will measure how much it takes to extract an entry using ExtractToFile.
+            ZipFile.CreateFromDirectory(currentTestFolderLocation, _archiveLocation, CompressionLevel, includeBaseDirectory: false);
+            _archive = ZipFile.OpenRead(_archiveLocation);
+            _entry = _archive.Entries[0];
+            _tempDestinationFileName = FileUtils.GetTestFilePath();
```

Comment:
Please let me know if this is the right way of getting a temporarily file name for concurrent writes/deletes. In long runs I was always getting "access to the path is denied errors", by long runs I mean this command:
```
dotnet run -c Release -f net6.0 
--filter *ExtractEntryToFile* --statisticalTest 3ms 
--coreRun 
  "C:\repos\runtime2\artifacts\bin\testhost\net6.0-windows-Release-x64\shared\Microsoft.NETCore.App\6.0.0\CoreRun.exe" 
  "C:\repos\runtime\artifacts\bin\testhost\net6.0-windows-Release-x64\shared\Microsoft.NETCore.App\6.0.0\CoreRun.exe" 
--maxIterationCount 101 --minIterationCount 100
```

Reply:
@adamsitnik there was another argument you used, but I can't find it. Do you remember what was it? Should David use it here too?

Reply:
ping @adamsitnik this was so long ago that I don't remember the context of the argument I mentioned 😄